nominal/core/dataset.py

```diff
@@ -135,6 +137,104 @@ def add_to_dataset_from_io(
+    ) -> Iterable[Channel]:
+        """Look up the metadata for all matching channels associated with this dataset.
+        NOTE: Provided channels may also be associated with other datasets-- use with caution.
+        Args:
+            exact_match: Filter the returned channels to those whose names match all provided strings (case insensitive).
+                For example, a channel named 'engine_turbine_rpm' would match against ['engine', 'turbine', 'rpm'],
+                whereas a channel named 'engine_turbine_flowrate' would not!
+            fuzzy_search_text: Filters the returned channels to those whose names fuzzily match the provided string.
+            channel_names: Filter the returned channels to those whose names exactly match the provided sequence of channel names (case sensitive).
+        Yields:
+            Yields a sequence of channel metadata objects which match the provided query parameters
+        """
+        next_page_token = None
+        while True:
+            query = datasource_api.SearchChannelsRequest(
+                data_sources=[self.rid],
+                exact_match=list(exact_match),
+                fuzzy_search_text=fuzzy_search_text,
+                previously_selected_channels={},
+                next_page_token=next_page_token,
+                page_size=None,
+                prefix=None,
+            )
+            response = self._clients.datasource.search_channels(self._clients.auth_header, query)
+            for channel_metadata in response.results:
+                if channel_names is not None and not channel_metadata.name in channel_names:
+                    continue
+
+                yield Channel._from_conjure(channel_metadata)
+
+            if response.next_page_token is None:
+                break
+            else:
+                next_page_token = response.next_page_token
+
+    def set_channel_units(self, channels_to_units: Mapping[str, str | None]) -> None:
+        """Set units for channels based on a provided mapping of channel names to units.
+        Args:
+            channels_to_units: A mapping of channel names to unit symbols.
+                NOTE: any existing units may be cleared from a channel by providing None as a symbol.
+        Raises:
+            ValueError: Unsupported unit symbol provided
+            conjure_python_client.ConjureHTTPError: Error completing requests.
+        """
+
+        # Get the set of all available unit symbols
+        available_units = set(_available_units(self._clients))
+
+        # Validate that all user provided unit symbols are valid
+        for channel_name, unit_symbol in channels_to_units.items():
+            if unit_symbol is None:
+                continue
+
+            if unit_symbol not in available_units:
+                raise ValueError(
+                    f"User provided unit {unit_symbol} for channel {channel_name} does not resolve to a unit recognized by nominal."
+                )
```

Comment:
it will be super ugly, but maybe worth it to print out the available units? e.g. unit <blah> not in recognized units <all units...>?

potentially, could just log it instead, idk.

Reply:
eh i think this might just be too long / annoying

Reply:
yea thats what I meant by ugly - not the code, just the rendering since there's so many units. lets remove it for now?

Reply:
ah thats a great idea